std/utf.d

```diff
@@ -4386,3 +4387,13 @@ if (isSomeChar!C)
     "𐐷".byUTF!wchar().equal([0xD801, 0xDC37]);
     "𐐷".byUTF!dchar().equal([0x00010437]);
 }
+
+///
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+    import std.exception : assertThrown;
+
+    assert("hello\xF0betty".byChar.byUTF!(dchar, UseReplacementDchar.yes).equal("hello\uFFFDetty"));
+    assertThrown!UTFException("hello\xF0betty".byChar.byUTF!(dchar, UseReplacementDchar.no).equal("hello\xF0betty"));
```

Comment:
This test is flawed, the `assectThrown` might be catching an exception thrown from `equal` (as it will be autodecoding its second parameter too).

Reply:
Perhaps use `each` or `walkLength` to expend the range instead.